python/ray/test/test_ray_init.py

```diff
@@ -0,0 +1,77 @@
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import os
+import pytest
+import redis
+
+import ray
+
+
+@pytest.fixture
+def start_ray_with_password():
+    ray.shutdown()
+
+    password = "some_password"
+    exception = None
+
+    # Fix for #3045
```

Comment:
Include the full URL

Reply:
Actually, I don't think we need to make a global variable, in generally we should be doing less in the fixture and more in the actual test. Why not move everything to the actual tests and just leave something like 

https://github.com/ray-project/ray/blob/6240ccbc6edeafd2e3223be1edb1448f3ea5d0ca/test/runtest.py#L205-L209

in the fixture?